accesshandler/pkg/runtime/lambda/revoke_grant.go

```diff
@@ -57,32 +56,34 @@ func (r *Runtime) RevokeGrant(ctx context.Context, grantID string, revoker strin
 		return nil, err
 	}
 
-	if grant.Status == types.ACTIVE {
-		err = prov.Provider.Revoke(ctx, string(grant.Subject), args)
-		if err != nil {
-			return nil, err
-		}
+	//do we need to check for grant status here?
```

Comment:
Looks to me that this check is in the wrong order.
You want to check the status of the workflow to determine if you need to run revoke or if we only need to "cancel" the existing workflow.

You need to change this around.